### PR TITLE
mate-desktop: some fixes, improvements, and new packages

### PR DESCRIPTION
**Pull request by romildo**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/mate/python-caja/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+
+  buildInputs = [
+    gtk3
+    caja
+    pythonPackages.python
+    pythonPackages.pygobject3
+  ];
+
+  configureFlags = [ "--with-cajadir=$$out/lib/caja/extensions-2.0" ];
```
Comment (dezgeg) [inline pkgs/desktops/mate/python-caja/default.nix]: Does this actually eventualy get expanded correctly? Usually it's done like this:
````
preConfigure = ''
    configureFlagsArray+=("--with-cajadir=$$out/lib/caja/extensions-2.0")
'';
````

Comment (romildo) [reply pkgs/desktops/mate/python-caja/default.nix]: It is working here and in other packages too.

```
$ nix-build --option build-use-chroot true -K -A mate.python-caja 2>&1 | tee /var/tmp/nix-build.log
[...]
Libraries have been installed in:
   /nix/store/72q7hql2lwwmgqm5jd3lr896mqkv53fq-python-caja-1.18.1/lib/caja/extensions-2.0
[...]
```

Comment (romildo) [reply pkgs/desktops/mate/python-caja/default.nix]: Searching for `'$$'` in nix files shows several other cases of using `'$$'` in `configureFlags`.

Comment (dezgeg) [reply pkgs/desktops/mate/python-caja/default.nix]: Ok, fair enough. But it really is a bad pattern to do, it's a completely unpredictable whether it's going to work or not, or even fail silently.

For example before commit 3d59a8e1a8341d09689d4ace8765ae134c264fe7 ncurses appeared to build fine, but the pkgconfig file was wrong, containing unexpanded `$(out)` / `$$out` etc.

**Comment by dezgeg** [comment]
Nice work!

File: pkgs/desktops/mate/python-caja/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/mate/python-caja/default.nix]: In the next updates I can abandon this bad pattern and follow your suggestion.

**Pull request merged**